synapse/util/caches/stream_change_cache.py

```diff
@@ -85,7 +85,7 @@ def has_entity_changed(self, entity: EntityType, stream_pos: int) -> bool:
         """Returns True if the entity may have been updated since stream_pos"""
```

Comment:
(Arbitrary spot to start a separate thread):

`_earliest_known_stream_pos` is used in a few places (screenshot includes the changes in this PR):

![image](https://user-images.githubusercontent.com/517124/201726349-8abd1845-fa5d-450d-a2ab-516fee727a61.png)

I'm a bit skeptical of the resulting consistency between strict and non-strict comparisons, but would like a double checking.

Reply:
More info for this -- this does *not* help with #14263. Maybe `has_any_entity_changed` needs to be strict `>`? That still doesn't fix that test though. 😢

Reply:
@erikjohnston describes this as:

> as if we call `entity_has_changed` for a stream pos then `has_entity_changed` *must* return True for that stream pos. *Either* because we remember the entity has changed or because the stream pos is before the minimum threshold